File: matlab/addData.m
function X = addData(x, sim)
[row, ~] = size(x);
X = sim * x;
for i = 1:row
    sumsim = sum(sim(i, :));
    if sumsim == 0
        continue
    end
    X(i, :) = X(i, :) / sumsim;
    fprintf("user %d add data finished!\n", i);
end
end

File: recommand-python/com/music/recommand/util/add_data.py
def addData(x, sim):
    X = sim * x
    row, column = sim.shape
    i = 0
    while i < row:
        sumsim = sim[i, :].sum()
        if sumsim == 0:
            i += 1
            continue
        X[i, :] /= sumsim
        i += 1
        print("add data finish: {:.6%}".format(i / row))
    return X